mempool/reactor.go

```diff
@@ -143,6 +167,18 @@ func (memR *MempoolReactor) broadcastTxRoutine(peer p2p.Peer) {
 				continue
 			}
 		}
+		// ensure peer hasn't already sent us this tx
+		skip := false
+		for i := 0; i < len(memTx.senders); i++ {
+			if peerID == memTx.senders[i] {
+				next = next.Next()
```

Comment:
I think it would be better to not do this here, since we're losing the NextWaitChain logic.

Instead, let's just figure out if skip is true (we should extract this to a hasSender function) and use:

```
if !skip{
  // send memTx
  ...
}

select {
 ...
} 
```